packages/sources/coinmetrics-test/src/endpoint/price-http.ts

```diff
@@ -0,0 +1,86 @@
+import { HttpTransport } from '@chainlink/external-adapter-framework/transports'
+import { ProviderResult } from '@chainlink/external-adapter-framework/util'
+import { VALID_QUOTES } from '../config'
+import { AssetMetricsEndpointTypes } from './price'
+
+export type MetricData = {
+  asset: string
+  time: string
+  ReferenceRateUSD?: string
+  ReferenceRateEUR?: string
+  ReferenceRateETH?: string
+  ReferenceRateBTC?: string
+}
+
+interface ResponseSchema {
+  data: MetricData[]
+  error?: {
+    type: string
+    message: string
+  }
+}
+
+type AssetMetricsHttpTypes = AssetMetricsEndpointTypes & {
+  Provider: {
+    RequestBody: never
+    ResponseBody: ResponseSchema
+  }
+}
+
+export const httpTransport = new HttpTransport<AssetMetricsHttpTypes>({
+  prepareRequests: (params, config) => {
+    return {
+      params,
+      request: {
+        baseURL: config.API_ENDPOINT,
+        url: `timeseries/asset-metrics`,
+        params: {
+          assets: [...new Set(params.map((p) => p.base.toUpperCase()))].join(','),
+          metrics: [...new Set(params.map((p) => `ReferenceRate${p.quote.toUpperCase()}`))].join(
+            ',',
+          ),
+          frequency: '1s',
+          api_key: config.API_KEY,
+          limit_per_asset: 1,
```

Comment:
Now that we're moving the http endpoint to batching do we need to worry about page size? According to coinmetrics docs, it defaults to 100. Unsure if we have that many unique bases coming into this adapter but to be safe should we set a high page size here? Max is 10,000

Reply:
I'll add the max allowed, and a warning if we get more